app/models/fs/api_connection.rb

```diff
@@ -72,9 +72,14 @@ def boxify
   private
 
   def generate_short_name_from_name(name)
-    generated_short_name = "FS" + name.split.map(&:first).join.upcase
-    boxes_with_same_short_name_count = tenant.boxes.where("short_name ~ ?", "#{generated_short_name}(%d)*").count
-    generated_short_name += boxes_with_same_short_name_count.to_s if boxes_with_same_short_name_count > 0
+    generated_base_short_name = "FS" + name.split.map(&:first).join.upcase
+    generated_short_name = generated_base_short_name
+
+    1.step do |i|
+      break if tenant.boxes.where("short_name ~ ?", "#{generated_short_name}(%d)*").empty?
+
+      generated_short_name = generated_base_short_name + i.to_s
```

Comment:
tu mi pride logickejsie vygenerovat to meno a skusit ci take v db uz je a potom uz nemusis nic lepit za tym, je to tak asi aj bezpecnejsie lebo sa nerozdeluje vyrabanie a checkovanie nazvu v dvoch roznych vetvach.

Reply:
Skusila som, pozri prosim, ci si myslel tak